modules/dxp/apps/analytics/analytics-reports-web/src/main/resources/META-INF/resources/js/AnalyticsReportsApp.js

```diff
@@ -9,73 +9,150 @@
  * distribution rights of the Software.
  */
 
-import React from 'react';
+import ClayAlert from '@clayui/alert';
+import ClayLoadingIndicator from '@clayui/loading-indicator';
+import {useIsMounted} from 'frontend-js-react-web';
+import {fetch} from 'frontend-js-web';
+import React, {useCallback, useEffect, useReducer} from 'react';
 
 import Navigation from './components/Navigation';
 import ConnectionContext from './context/ConnectionContext';
 import {StoreContextProvider} from './context/store';
-import APIService from './utils/APIService';
 
 import '../css/analytics-reports-app.scss';
 
-export default function ({context, props}) {
-	const {languageTag, namespace, page} = context;
-	const {defaultTimeRange, defaultTimeSpanKey, timeSpans} = context;
-	const {validAnalyticsConnection} = context;
-
-	const {
-		authorName,
-		authorPortraitURL,
-		authorUserId,
-		publishDate,
-		title,
-	} = props;
-	const {trafficSources} = props;
-
-	const {
-		getAnalyticsReportsHistoricalReadsURL,
-		getAnalyticsReportsHistoricalViewsURL,
-		getAnalyticsReportsTotalReadsURL,
-		getAnalyticsReportsTotalViewsURL,
-	} = context.endpoints;
-
-	const api = APIService({
-		endpoints: {
-			getAnalyticsReportsHistoricalReadsURL,
-			getAnalyticsReportsHistoricalViewsURL,
-			getAnalyticsReportsTotalReadsURL,
-			getAnalyticsReportsTotalViewsURL,
+const dataReducer = (state, action) => {
+	switch (action.type) {
+		case 'LOAD_DATA':
+			return {
+				...state,
+				loading: true,
+			};
+
+		case 'SET_ERROR':
+			return {
+				...state,
+				error: action.error,
+				loading: false,
+			};
+
+		case 'SET_DATA':
+			return {
+				data: action.data,
+				error: action.data?.error,
+				loading: false,
+				publishedToday: action.publishedToday,
```

Comment:
Changed [here](https://github.com/liferay-frontend/liferay-portal/pull/287/commits/b1052c97fbaaf445066adeae552c07278c179776).

Reply:
Why are you still adding `publishedToday` as part of the state if it's already present in `data`?

Also I'm repeating myself: 
> doing here the calculation
new Date().toDateString() === new Date(data.context.publishDate).toDateString(), instead of in the fetch response, so you have all the state creation in the same place (the reducer)

It's your call of course, just to be sure you read it

Reply:
I'm not sure what you mean 🤔  `publishedToday` is not present in the `data`. In the `data` we have `publishDate` and I need to know if that publish date is today or not doing that calculation:
```js
new Date().toDateString() === new Date(data.context.publishDate).toDateString()
```

Can you be more specific, please? Thanks.

Reply:
Outdated code, you removed `publishedToday: action.publishedToday,` from line 44 but I was still seeing it. So that part ok.

> doing here the calculation
new Date().toDateString() === new Date(data.context.publishDate).toDateString(), instead of in the fetch response, so you have all the state creation in the same place (the reducer)

About that I mean that you can place that calc in the reducer so everything is in the same place

```
const dataReducer = (state, action) => {
    ...
    case 'SET_DATA':
        return {
            data: {
                ...action.data,
                publishedToday: new Date().toDateString() ===
                    new Date(
                        action.data?.publishDate
                    ).toDateString(),
            },
            error: action.data?.error,
            loading: false,
        };

...

safeDispatch({
    data: data.context,
    type: 'SET_DATA',
})
``` 

but that's as you prefer.

Reply:
I'm going to change it with your proposal :) Do I need the `?` in `action.data?.publishDate`? 🤔

Reply:
Do you need `?` in `action.data?.error`? It's the same, it depends on what you want/need to safe guard ;)

Reply:
Ok, it's true, I'm going to update the pull with the last changes (from your feedback), thanks!